packages/webapp/pages/[userId]/index.tsx

```diff
@@ -1,306 +1,104 @@
-import PostsSection from '@dailydotdev/shared/src/components/profile/PostsSection';
-import AuthorStats from '@dailydotdev/shared/src/components/profile/AuthorStats';
-import { Dropdown } from '@dailydotdev/shared/src/components/fields/Dropdown';
-import { useViewSize, ViewSize } from '@dailydotdev/shared/src/hooks';
-import { weeklyGoal } from '@dailydotdev/shared/src/lib/constants';
-import { ButtonSize } from '@dailydotdev/shared/src/components/buttons/ButtonV2';
-import CalendarHeatmap from '../../components/CalendarHeatmap';
+import { RanksWidget } from '@dailydotdev/shared/src/components/profile/RanksWidget';
+import { useActivityTimeFilter } from '@dailydotdev/shared/src/hooks/profile/useActivityTimeFilter';
+import { ReadingTagsWidget } from '@dailydotdev/shared/src/components/profile/ReadingTagsWidget';
+import { ReadingHeatmapWidget } from '@dailydotdev/shared/src/components/profile/ReadingHeatmapWidget';
+import {
+  generateQueryKey,
+  RequestKey,
+} from '@dailydotdev/shared/src/lib/query';
+import { getProfile } from '@dailydotdev/shared/src/lib/user';
 import {
   getLayout as getProfileLayout,
-  getStaticProps as getProfileStaticProps,
   getStaticPaths as getProfileStaticPaths,
+  getStaticProps as getProfileStaticProps,
   ProfileLayoutProps,
-} from '../../components/layouts/ProfileLayout';
-
-export const getStaticProps = getProfileStaticProps;
-export const getStaticPaths = getProfileStaticPaths;
-
-const readHistoryToValue = (value: UserReadHistory): number => value.reads;
-const readHistoryToTooltip = (
-  value: UserReadHistory,
-  date: Date,
-): ReactNode => {
-  const formattedDate = date.toLocaleString('en-US', {
-    month: 'short',
-    day: '2-digit',
-    year: 'numeric',
-  });
-  if (!value?.reads) {
-    return `No posts read on ${formattedDate}`;
-  }
-  return (
-    <>
-      <strong>
-        {value.reads} article{value.reads > 1 ? 's' : ''} read
-      </strong>
-      &nbsp;on {formattedDate}
-    </>
+} from '../../components/layouts/ProfileLayout/v2';
+import { Readme } from '../../../shared/src/components/profile/Readme';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const ProfilePage = ({
+  user: initialUser,
+}: ProfileLayoutProps): ReactElement => {
+  const { user: loggedUser, tokenRefreshed } = useContext(AuthContext);
+  const isSameUser = loggedUser?.id === initialUser?.id;
+
+  const {
+    selectedHistoryYear,
+    setSelectedHistoryYear,
+    before,
+    after,
+    yearOptions,
+    fullHistory,
+  } = useActivityTimeFilter();
+
+  const { data: user } = useQuery(
+    generateQueryKey(RequestKey.Profile, initialUser),
+    () => getProfile(initialUser?.id),
+    {
+      placeholderData: initialUser,
+      refetchOnReconnect: false,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      // Make sure the logged in user's profile is always up to date
+      enabled: !!initialUser && isSameUser,
```

Comment:
This would just mean it will always refetch even if you didn't change, not needed for this PR, but actually had a thought about it.

1. Can we not manually manipulate the cache as we do in other places, that should work I think
2. If issues we could actually start using back redirects with query params like ?r=true as when `r` is set we refresh enable?

Although think we prefer 1 with a staleTime?

Reply:
This will not fix all the caching issues I'm trying to solve. The primary one is given we use next ISR, the page is cached for quite some time. But let's say I just updated my readme and then refresh the page, without this I'll see the old readme. With this I see always fresh data

Reply:
Ah yes, but in that case would the 2nd option not work of leveraging query params to force refreshes?
(it's not really blocking, more opening the discussion for future things as we have quite a few of these issues already)

Reply:
When the user manually decides to visit their profile again I cannot force them into adding query param, right? Or am I missing something?

Reply:
That case no, my suggestion was:
- user edit profile
- on save we invoke toast + redirect back to profile + ?r=true

(but now I'm not sure if we didn't redirect for specific reason maybe)

Reply:
I feel like this is not a coherent solution for my use case then. If that's ok can we keep it as is for now?

Reply:
Yeah 100% ok for this PR :)